diff -ru chef-client/files/default/tests/minitest/cron_test.rb chef-client.new/files/default/tests/minitest/cron_test.rb
--- chef-client/files/default/tests/minitest/cron_test.rb	2014-03-18 22:54:21.000000000 +0000
+++ chef-client.new/files/default/tests/minitest/cron_test.rb	2014-03-25 13:10:03.820231638 +0000
@@ -30,8 +30,7 @@
     if node['chef_client']['cron']['use_cron_d']
       file('/etc/cron.d/chef-client').must_match %r{/bin/sleep \d+; (([A-Za-z]+=.*)?) /usr/bin/chef-client > /dev/null 2>&1}
     else
-      cron('chef-client').command
-        .must_match %r{/bin/sleep \d+; (([A-Za-z]+=.*)?)|[\s] /usr/bin/chef-client > /dev/null 2>&1}
+      cron('chef-client').command.must_match %r{/bin/sleep \d+; (([A-Za-z]+=.*)?)|[\s] /usr/bin/chef-client > /dev/null 2>&1}
     end
   end
 end
