--- client.py.orig      2013-11-28 09:34:01.095679290 -0500
+++ client.py   2013-11-28 09:36:04.117276915 -0500
@@ -30,6 +30,7 @@
                            tenant_name=kwargs.get('tenant_name'),
                            auth_url=kwargs.get('auth_url'),
                            region_name=kwargs.get('region_name'),
+                           cacert=kwargs.get('cacert'),
                            insecure=kwargs.get('insecure'))


@@ -72,6 +73,7 @@
             'region_name': kwargs.get('os_region_name'),
             'service_type': kwargs.get('os_service_type'),
             'endpoint_type': kwargs.get('os_endpoint_type'),
+            'cacert': kwargs.get('os_cacert'),
             'insecure': kwargs.get('insecure'),
         }
         _ksclient = _get_ksclient(**ks_kwargs)

--- shell.py.orig       2013-11-28 09:34:12.030487491 -0500
+++ shell.py    2013-11-28 09:35:31.491828234 -0500
@@ -165,6 +165,12 @@
         parser.add_argument('--os_endpoint_type',
                             help=argparse.SUPPRESS)

+        parser.add_argument('--os-cacert',
+                            default=utils.env('OS_CACERT'),
+                            help="Specify a CA bundle file to use in "
+                            "verifying a TLS (https) server certificate. "
+                            "Defaults to env[OS_CACERT]")
+
         return parser

     def get_subcommand_parser(self, version):
