--- shell.py.orig       2013-11-28 09:21:18.549667830 -0500
+++ shell.py    2013-11-28 09:31:03.286806741 -0500
@@ -166,6 +166,12 @@
         parser.add_argument('--os_endpoint_type',
                             help=argparse.SUPPRESS)

+        parser.add_argument('--os-cacert',
+                            default=utils.env('OS_CACERT'),
+                            help="Specify a CA bundle file to use in "
+                            "verifying a TLS (https) server certificate. "
+                            "Defaults to env[OS_CACERT]")
+
         parser.add_argument('-t', '--token-only',
                             default=bool(False),
                             action='store_true',
@@ -220,6 +226,7 @@
                                tenant_id=kwargs.get('tenant_id'),
                                tenant_name=kwargs.get('tenant_name'),
                                auth_url=kwargs.get('auth_url'),
+                               cacert=kwargs.get('cacert'),
                                insecure=kwargs.get('insecure'))

     def _get_endpoint(self, client, **kwargs):
@@ -296,6 +303,7 @@
             'auth_url': args.os_auth_url,
             'service_type': args.os_service_type,
             'endpoint_type': args.os_endpoint_type,
+            'cacert': args.os_cacert,
             'insecure': args.insecure
         }
