--- shell.py    2014-03-02 17:38:31.000000000 -0500
+++ shell-new.py        2014-07-12 14:34:55.473819753 -0400
@@ -83,6 +83,7 @@
                             ' prepended to your cert file.')

         parser.add_argument('--ca-file',
+                            default=utils.env('OS_CACERT'),
                             help='Path of CA SSL certificate(s) used to verify'
                             ' the remote server\'s certificate. Without this'
                             ' option the client looks'
@@ -244,6 +245,7 @@
         :param token: token to use instead of username/password
         """
         kc_args = {'auth_url': kwargs.get('auth_url'),
+                   'cacert': kwargs.get('cacert'),
                    'insecure': kwargs.get('insecure')}

         if kwargs.get('tenant_id'):
@@ -343,6 +345,7 @@
                                        " env[OS_AUTH_URL]")

         kwargs = {
+            'cacert': args.ca_file,
             'username': args.os_username,
             'password': args.os_password,
             'token': args.os_auth_token,
