--- shell.py.orig       2013-11-21 12:59:51.046409233 -0500
+++ shell.py    2013-11-21 13:12:49.210385762 -0500
@@ -74,6 +74,7 @@
                             ' prepended to your cert file.')

         parser.add_argument('--ca-file',
+                            default=utils.env('OS_CACERT'),
                             help='Path of CA SSL certificate(s) used to verify'
                             ' the remote server\'s certificate. Without this'
                             ' option the client looks'
@@ -220,6 +221,7 @@
                                tenant_id=kwargs.get('tenant_id'),
                                tenant_name=kwargs.get('tenant_name'),
                                auth_url=kwargs.get('auth_url'),
+                               cacert=kwargs.get('cacert'),
                                insecure=kwargs.get('insecure'))

     def _get_endpoint(self, client, **kwargs):
@@ -296,6 +298,7 @@
             'auth_url': args.os_auth_url,
             'service_type': args.os_service_type,
             'endpoint_type': args.os_endpoint_type,
+            'cacert': args.ca_file,
             'insecure': args.insecure
         }
